### PR TITLE
add generic jdbc impl of a key value store

### PR DESCRIPTION
**Pull request by tjwilson90**

### CODE REVIEW

**Comment by tjwilson90** [comment]
@carrino @rjullman

File: atlasdb-jdbc/src/main/java/com/palantir/atlasdb/keyvalue/jdbc/JdbcKeyValueService.java [hunk 1]
```diff
@@ -0,0 +1,955 @@
...
+                        field(ROW_NAME, byte[].class),
+                        field(COL_NAME, byte[].class),
+                        field(TIMESTAMP, Long.class),
+                        field(VALUE, byte[].class));
+        query = batch.addValuesForInsert(query);
+        try {
+            query.execute();
+        } catch (DataAccessException e) {
+            if (allowReinserts) {
+                Result<? extends Record> records = ctx
```
Comment (carrino) [inline atlasdb-jdbc/src/main/java/com/palantir/atlasdb/keyvalue/jdbc/JdbcKeyValueService.java]: it's hard to tell if this handles the idempotent put case.

Comment (carrino) [inline atlasdb-jdbc/src/main/java/com/palantir/atlasdb/keyvalue/jdbc/JdbcKeyValueService.java]: NM, i see the putbatch.getNextBatch handles that.

File: atlasdb-jdbc/src/main/java/com/palantir/atlasdb/keyvalue/jdbc/JdbcKeyValueService.java [hunk 2]
```diff
@@ -0,0 +1,955 @@
...
+            }
+        };
+        return ClosableIterators.wrap(iter.iterator());
+    }
+
+    @Override
+    public ClosableIterator<RowResult<Set<Value>>> getRangeWithHistory(String tableName,
+                                                                       RangeRequest rangeRequest,
+                                                                       long timestamp) {
+        throw new UnsupportedOperationException();
```
Comment (carrino) [inline atlasdb-jdbc/src/main/java/com/palantir/atlasdb/keyvalue/jdbc/JdbcKeyValueService.java]: no TieredKeyValueService support!

**Pull request merged**